README.md

```diff
@@ -84,14 +87,138 @@ const searchConfig = new SearchConfig({
 | **i18n** | Object | undefined | Override translations | No |
 
 #### Currently supported searchTypes
+
 ```jsx
 const searchTypes = ['AudioObject', 'DigitalDocument', 'Person', 'MusicComposition', 'VideoObject'];
 ```
 
+#### Custom searchTypes
+
+A custom search type can also be used. This can be either a Class or Object with at least the following properties:
+name, filters, searchAllQuery and searchQuery.
+
+For example to create a custom search type for SoftwareApplications;
+
+```js
+import gql from 'graphql-tag';
+
+class CustomType {
+  static name = 'SoftwareApplication';
+
+  static filters = [];
+
+  static searchAllQuery = gql`
+    query($query: String!, $first: Int = 9999) {
+      allResults: searchMetadataText(onTypes: [SoftwareApplication], onFields: [title], substring: $query, first: $first) {
+        ... on SoftwareApplication {
+          identifier
+          format
+          _searchScore
+        }
+      }
+    }
+  `;
+
+  static searchQuery = gql`
+    query($filter: _SoftwareApplicationFilter) {
+      results: SoftwareApplication(filter: $filter, first: 50) {
+        __typename
+        ... on SoftwareApplication {
+          identifier
+          name
+          title
+          creator
+          source
+        }
+      }
+    }
+  `;
+}
+```
+
+If your setup doesn't
+support [static class properties](https://babeljs.io/docs/en/babel-plugin-proposal-class-properties), you can define the
+properties like this as well:
+
+```js
+class CustomType {
+}
+
+// or const CustomType = {}
+
+CustomType.name = 'SoftwareApplication';
+CustomType.filters = [];
+CustomType.searchAllQuery = gql`
+    query($query: String!, $first: Int = 9999) {
+      allResults: searchMetadataText(onTypes: [SoftwareApplication], onFields: [title], substring: $query, first: $first) {
+        ... on SoftwareApplication {
+          identifier
+          format
+          _searchScore
+        }
+      }
+    }
+  `;
+
+CustomType.searchQuery = gql`
+    query($filter: _SoftwareApplicationFilter) {
+      results: SoftwareApplication(filter: $filter, first: 50) {
+        __typename
+        ... on SoftwareApplication {
+          identifier
+          name
+          title
+          creator
+          source
+        }
+      }
+    }
+  `;
+```
+
+#### Custom result
+
+If you want to change the rendered result, you can use the `renderSearchResult` prop to use a custom function. This function should return a valid JSX object.
+
+```jsx
+import React, { Component } from 'react'
+import MultiModalComponent, { SearchConfig, searchTypes, SearchResult } from 'trompa-multimodal-component'
+
+const searchConfig = new SearchConfig({
+  searchTypes: [searchTypes.MusicComposition],
+});
+
+const renderSearchResult = (type, item, onClick) => {
+  // You can return custom JSX
+  // return (
+  //   <div onClick={onClick}>{item.title}</div>
+  // );
+
+  return <SearchResult title={item.title} variant="default" onClick={onClick} />
```

Comment:
maybe a comment here to say that your renderSearchResult should always finish with this to handle the default objects

Reply:
yes, good suggestion!